feature_layers/display-subtype-feature-layer/src/main/java/com/esri/samples/feature_layers/DisplaySubtypeFeatureLayerController.java

```diff
@@ -0,0 +1,148 @@
+      // display the current map scale 
+      mapView.addMapScaleChangedListener(mapScaleChangedEvent ->
+        currentMapScaleLabel.setText("Current map scale: 1:" + Math.round(mapView.getMapScale())));
+      
+      // set the viewpoint to Naperville, Illinois
+      Viewpoint initialViewpoint = new Viewpoint(new Envelope(-9812691.11079696, 5128687.20710657,
+        -9812377.9447607, 5128865.36767282, SpatialReferences.getWebMercator()));
+      map.setInitialViewpoint(initialViewpoint);
+
+      // create a subtype feature layer from the service feature table, and add it to the map
+      ServiceFeatureTable serviceFeatureTable = new ServiceFeatureTable("https://sampleserver7.arcgisonline" +
+        ".com/arcgis/rest/services/UtilityNetwork/NapervilleElectric/FeatureServer/100");
+      SubtypeFeatureLayer subtypeFeatureLayer = new SubtypeFeatureLayer(serviceFeatureTable);
+      map.getOperationalLayers().add(subtypeFeatureLayer);
+      
+      // access the json required for sublayer label definitions
+      File jsonFile = new File(System.getProperty("data.dir"),
+          Objects.requireNonNull(getClass().getClassLoader().getResource("label_definition.json")).getFile());
+      final String json;
+      // read in the complete file as a string
+      try (Scanner scanner = new Scanner(jsonFile)) {
+        json = scanner.useDelimiter("\\A").next();
+      }
+
+      // load the subtype feature layer
+      subtypeFeatureLayer.loadAsync();
+      subtypeFeatureLayer.addDoneLoadingListener(() -> {
+        
+        // get the Street Light sublayer and define its labels
+        sublayer = subtypeFeatureLayer.getSublayerWithSubtypeName("Street Light");
+        sublayer.setLabelsEnabled(true);
+        sublayer.getLabelDefinitions().add(LabelDefinition.fromJson(json));
+
+        // get the original renderer of the sublayer (white and black circular icon)
+        originalRenderer = sublayer.getRenderer();
+        // create a custom renderer for the sublayer (light pink diamond symbol)
+        Symbol symbol = new SimpleMarkerSymbol(SimpleMarkerSymbol.Style.DIAMOND, 0xfff58f84, 20);
+        alternativeRenderer = new SimpleRenderer(symbol);
+
+      });
+
+    } catch (Exception e) {
+      // on any error, display the stack trace.
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Sets the minimum scale of the labels for the sublayer.
+   */
+  @FXML
+  private void handleMinScaleButtonClicked() {
+    sublayer.setMinScale(mapView.getMapScale());
+    minScaleLabel.setText("Sublayer min scale: 1:" + Math.round(sublayer.getMinScale()));
+  }
+
+  /**
+   * Sets the visibility of the sublayer.
+   */
+  @FXML
+  private void controlSublayerVisibility() {
+    sublayer.setVisible(sublayerVisibilityCheckbox.isSelected());
+  }
+  
+  /**
+   * Sets the renderer of the sublayer to its original format (a white and black circular icon). 
+   */
+  @FXML
+  private void handleOriginalRendererButtonClicked() {
```

Comment:
This is okay for two radio buttons. With more, you'd probably set `userData` on the radio button, and then watch the toggle group for changes for the selected user data.

Reply:
Good to know thanks! I was wondering if there was a better way to handle this.

Reply:
nvm, what you're doing is fine